frontend/src/bundles/common/components/button/button.tsx

```diff
@@ -1,10 +1,75 @@
-type Properties = {
-    label: string;
-    type?: 'button' | 'submit';
+import clsx from 'clsx';
+
+import {
+    type ButtonTheme,
+    ButtonSize,
+    IconName,
+    IconSize,
+} from '~/bundles/common/enums/enums';
+import { type ButtonType, type ValueOf } from '~/bundles/common/types/types';
+
+import { Icon } from '../icon/icon';
+import styles from './styles.module.scss';
+
+type ButtonProperties = {
+    label?: string;
+    type?: ButtonType;
+    theme?: ValueOf<typeof ButtonTheme>;
+    size?: ValueOf<typeof ButtonSize>;
+    onClick?: React.MouseEventHandler<HTMLButtonElement>;
+    isDisabled?: boolean;
+    className?: string;
+    isRightIcon?: boolean;
+    isLeftIcon?: boolean;
 };
 
-const Button: React.FC<Properties> = ({ type = 'button', label }) => (
-    <button type={type}>{label}</button>
-);
+const Button: React.FC<ButtonProperties> = ({
+    label,
+    onClick,
+    type = 'button',
+    theme,
+    size = ButtonSize.MEDIUM,
+    isDisabled = false,
+    className,
+    isRightIcon,
+    isLeftIcon,
+}) => {
+    return (
+        <button
+            disabled={isDisabled}
+            onClick={onClick}
+            className={clsx(
+                styles.btn,
+                styles[`btn__${size}`],
```

Comment:
Maybe it's just my quirk, but I don't like the way this code looks. I'd prefer to see something like this
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/assets/80532461/7e451cbe-3fcd-4135-8c37-c767ecd8b950)
Then in layout it looks less cumbersome
![image](https://github.com/BinaryStudioAcademy/bsa-winter-2023-2024-resumegemm/assets/80532461/22845cbf-6bc6-40ca-99ff-e50963cb02dd)